mne/morph.py

```diff
@@ -0,0 +1,1443 @@
+
+    See Also
+    --------
+    stc.morph : Directly morph SourceEstimate or VectorSourceEstimate.
+    stc.as_volume : Convert VolSourceEstimate into NIfTI.
+    Example : :ref:`sphx_glr_auto_examples_inverse_plot_morph_volume_stc.py`.
+    Example : :ref:`sphx_glr_auto_examples_inverse_plot_morph_surface_stc.py`.
+    Tutorial : :ref:`sphx_glr_auto_tutorials_plot_morph_stc.py`.
+    """
+
+    def __init__(self, subject_from=None, subject_to='fsaverage',
+                 subjects_dir=None, src=None, niter_affine=(100, 100, 10),
+                 niter_sdr=(5, 5, 3), spacing=5, smooth=None, warn=True,
+                 xhemi=False, precomputed=None, verbose=False):
+        super(SourceMorph, self).__init__()
+        # it's impossible to use the class without passing this check, so it
+        # only needs to be checked here
+        if (not check_version('nibabel', '2.1.0') or
+                not check_version('dipy', '0.10.1')):
+            raise ImportError(
+                'NiBabel 2.1.0 and DiPy 0.10.1 or higher must be correctly '
+                'installed and accessible from Python')
+
+        if src is not None and not isinstance(src, SourceSpaces):
+            raise ValueError('src must be an instance of SourceSpaces or a '
+                             'path to a saved instance of SourceSpaces')
+        # Params
+        self.kind = 'surface' if src is None else src.kind
+        self.subject_from = _check_subject_from(subject_from, src)
+        self.subject_to = subject_to
+        self.subjects_dir = subjects_dir
+
+        # Params for volumetric morphing
+        self.niter_affine = niter_affine
+        self.niter_sdr = niter_sdr
+
+        self.spacing = spacing
+
+        # Params for surface morphing
+        self.smooth = smooth
+        self.warn = warn
+        self.xhemi = xhemi
+
+        self.params = dict()
+
+        # apply precomputed data and return
+        if precomputed is not None:
+            self._update_morph_data(precomputed)
+            return
+
+        # get data to perform morph and as_volume
+        if src is not None:
+            self._update_morph_data(_get_src_data(src))
+            self._compute_morph_data(verbose=verbose)
+
+    def __call__(self, stc_from, as_volume=False, mri_resolution=False,
+                 mri_space=False, apply_morph=True, format='nifti1',
+                 verbose=None):
+        """Morph data.
+
+        Parameters
+        ----------
+        stc_from : VolSourceEstimate | SourceEstimate | VectorSourceEstimate
+            The source estimate to morph.
+        as_volume : bool
+            Whether to output a NIfTI volume. stc_from has to be a
+            VolSourceEstimate. Default is False.
+        mri_resolution: bool | tuple | int | float
+            If True the image is saved in MRI resolution. Default False.
+            WARNING: if you have many time points the file produced can be
+            huge.
```

Comment:
better to put this warning in the `save` method, no?

Reply:
I actually wouldn't like to put it to the save method, because it will become huge here, not just when saving (it's huge also in working memory)... ?